src/Storages/Cache/ExternalDataSourceCache.cpp

```diff
@@ -94,6 +94,8 @@ bool RemoteReadBuffer::nextImpl()
         return status;
     }
 
+    //file_buffer::pos should increase correspondingly when RemoteReadBuffer is consumed, otherwise start_offset will be incorrect.
+    local_file_holder->file_buffer->position() = local_file_holder->file_buffer->buffer().begin() + BufferBase::offset();
```

Comment:
Can you please add a test for that which previously failed? in `Hive` storage tests. See `tests/integration/test_storage_hive/test.py`

Reply:
This is not a case that can be reproduced stably.